src/librustc_mir/dataflow/generic/mod.rs

```diff
@@ -166,6 +168,22 @@ pub trait Analysis<'tcx>: AnalysisDomain<'tcx> {
         args: &[mir::Operand<'tcx>],
         return_place: &mir::Place<'tcx>,
     );
+
+    /// Creates an `Engine` to find the fixpoint for this dataflow problem.
+    ///
+    /// This is functionally equivalent to calling the appropriate `Engine` constructor. It should
+    /// not be overridden. Its purpose is to allow consumers of this API to use method-chaining.
```

Comment:
nit: it is a little weird to make the statement "it should not be overridden", and in the same commit, include an overriding (i.e. non-default) implementation.

I do understand what's going on, i just couldn't help but point it out. (I'd probably be satisfied if it said "it should not be overridden apart from specialized instances such as in `GenKillAnalysis`.")

Reply:
That's true. I can do better here.